### ISSUE TITLE
minidlna does not index all files

### ISSUE THREAD

**Issue by djaenecke** [azatoth/minidlna#16]
I have a directory with 2377 mp3 files, but minidlna will only index 799 of them. There are no errors in the log for specific files (log level warn). There is this SQL error but I am not sure if it is related, since it is shown after scanning has finished:
```
sql.c:117: warn: sql_get_int_field: step failed: SQL logic error or missing database
SELECT (select count(distinct DETAIL_ID) from OBJECTS o left join DETAILS d on (o.DETAIL_ID = d.ID) where (OBJECT_ID glob '*$*') and (o.CLASS like "item.audioItem%" and REF_ID is NULL)) + (select count(*) from OBJECTS o left join DETAILS d on (o.DETAIL_ID = d.ID) where (OBJECT_ID = '*') and (o.CLASS like "item.audioItem%" and REF_ID is NULL))
```
All files and directories are readable by the minidlna user.
The problem occures in 1.1.5 and 1.2.1.

Any idea how to find out why these files are ignored (and how to fix it)?

**Comment by VBota1** [comment]
I found out that it has a problems with some file names. I think it's a bug but I'm not sure on what.

**Comment by mu88** [comment]
@VBota1 did you ever solve this issue? I'm also having trouble with files containing special chars like `[` or `ä`.

I'm using `1.0.2.r43` on a ASUSTOR NAS

**Comment by VBota1** [comment]
I did not solve it. I just wrote a script and renamed the files. I'm no longer using Minidlna (I don't need it anymore).

**Comment by Spider84** [comment]
Can anyone post here some examples of filenames to check this bug?
I can try to fix it.

**Comment by mu88** [comment]
This is my test file `test7.m3u` which does not contain any items in MiniDLNA:
```m3u
#EXTM3U
#EXTINF:196,Dean Lewis - Be Alright [Explicit]
#EXTVLCOPT:file-caching=1000
Dean%20Lewis/Be%20Alright/01-01-Be%20Alright.mp3
#EXTINF:196,Dean Lewis - Be Alright
#EXTVLCOPT:file-caching=1000
Dean%20Lewis/Be%20Alright/01-01-Be%20Alright.mp3
#EXTINF:196,Dean Lewis - Be Älright [Explicit]
#EXTVLCOPT:file-caching=1000
Dean%20Lewis/Be%20Alright/01-01-Be%20Älright.mp3
#EXTINF:196,Dean Lewis - Be Älright
#EXTVLCOPT:file-caching=1000
Dean%20Lewis/Be%20Alright/01-01-Be%20Älright.mp3
```

**Comment by VBota1** [comment]
Ștefan Hrușcă - Rugă pentru părinți
Ștefan Hrușcă - La săvârșitu' lumii
Pasărea Phoenix - Țiganiada 
Béla Kamocsa
Zoly Kovács
Totuși sînt ca voi sau Floarea stîncilor (EP, vinil 7", 1969)
Tuareg/Mr. G's Promises single, vinil 7", 1988
Ciocîrlia/Perestroika (single/promo, vinil 7", 1990)
Vino, Țepeș! album de studio, CD, 2014

**Comment by Spider84** [comment]
> This is my test file `test7.m3u` which does not contain any items in MiniDLNA:
> 
> ```
> #EXTM3U
> #EXTINF:196,Dean Lewis - Be Alright [Explicit]
> #EXTVLCOPT:file-caching=1000
> Dean%20Lewis/Be%20Alright/01-01-Be%20Alright.mp3
> ```
This file too? O_o

**Comment by Spider84** [comment]
Hi. Can you send me some files? I can't reproduce the bug.
https://spider_vc.t.me/ or email: 🕷<at>🕷.vc   (spider 😁)

**Comment by mu88** [comment]
> > This is my test file `test7.m3u` which does not contain any items in MiniDLNA:
> > ```
> > #EXTM3U
> > #EXTINF:196,Dean Lewis - Be Alright [Explicit]
> > #EXTVLCOPT:file-caching=1000
> > Dean%20Lewis/Be%20Alright/01-01-Be%20Alright.mp3
> > ```
> 
> This file too? O_o

Yes, this one fails, too :( I'm not on Telegram, but I wrote you an email - hope you received it 🤞🏻

**Comment by Spider84** [comment]
Ok
